File: mobile/src/main/java/com/siliconlabs/bledemo/Bluetooth/BLE/BluetoothStateReceiver.java
package com.siliconlabs.bledemo.Bluetooth.BLE;

import android.bluetooth.BluetoothAdapter;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

/**
 * @author Comarch S.A.
 */
public class BluetoothStateReceiver extends BroadcastReceiver {

    private final BluetoothStateListener bluetoothStateListener;


    public BluetoothStateReceiver(BluetoothStateListener bluetoothStateListener) {
        this.bluetoothStateListener = bluetoothStateListener;
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent.getAction();
        if (action == null) {
            return;
        }
        if (action.equals(BluetoothAdapter.ACTION_STATE_CHANGED)) {
            int state = intent.getIntExtra(BluetoothAdapter.EXTRA_STATE, BluetoothAdapter.ERROR);

            switch (state) {
                case BluetoothAdapter.ERROR:
                case BluetoothAdapter.STATE_OFF:
                case BluetoothAdapter.STATE_TURNING_OFF:
                    notifyState(false);
                    break;
                case BluetoothAdapter.STATE_ON:
                    notifyState(true);
                    break;
            }
        }
    }

    private void notifyState(boolean enabled) {
        bluetoothStateListener.onBluetoothStateChanged(enabled);
    }

    public interface BluetoothStateListener {
        void onBluetoothStateChanged(boolean enabled);
    }
}

File: mobile/src/main/java/com/siliconlabs/bledemo/Bluetooth/BLE/BluetoothStateReceiver.kt
package com.siliconlabs.bledemo.bluetooth.ble

import android.bluetooth.BluetoothAdapter
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

/**
 * @author Comarch S.A.
 */
class BluetoothStateReceiver(private val bluetoothStateListener: BluetoothStateListener) : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action ?: return
        if (action == BluetoothAdapter.ACTION_STATE_CHANGED) {

            val state = intent.getIntExtra(BluetoothAdapter.EXTRA_STATE, BluetoothAdapter.ERROR)
            when (state) {
                BluetoothAdapter.ERROR, BluetoothAdapter.STATE_OFF, BluetoothAdapter.STATE_TURNING_OFF -> notifyState(false)
                BluetoothAdapter.STATE_ON -> notifyState(true)
            }
        }
    }

    private fun notifyState(enabled: Boolean) {
        bluetoothStateListener.onBluetoothStateChanged(enabled)
    }

    interface BluetoothStateListener {
        fun onBluetoothStateChanged(enabled: Boolean)
    }

}